internal/service/bedrockagent/agent_versions_data_source.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package bedrockagent
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/bedrockagent"
+	awstypes "github.com/aws/aws-sdk-go-v2/service/bedrockagent/types"
+	"github.com/hashicorp/terraform-plugin-framework-timetypes/timetypes"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework/flex"
+	fwtypes "github.com/hashicorp/terraform-provider-aws/internal/framework/types"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @FrameworkDataSource("aws_bedrockagent_agent_versions, name="Agent Versions")
+func newDataSourceAgentVersions(context.Context) (datasource.DataSourceWithConfigure, error) {
+	return &dataSourceAgentVersions{}, nil
+}
+
+const (
+	DSNameAgentVersions = "Agent Versions Data Source"
+)
+
+type dataSourceAgentVersions struct {
+	framework.DataSourceWithConfigure
+}
+
+func (d *dataSourceAgentVersions) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) { // nosemgrep:ci.meta-in-func-name
+	resp.TypeName = "aws_bedrockagent_agent_versions"
+}
+
+func (d *dataSourceAgentVersions) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"agent_id": schema.StringAttribute{
+				Required: true,
+			},
+			"agent_version_summaries": schema.ListAttribute{
```

Comment:
To work properly with v5 of the Terraform protocol (pre-1.0 versions of Terraform) this will have to be converted to a `ListNestedBlock`.